libproc/lustre.c

```diff
@@ -131,6 +131,42 @@ typedef enum {
  * On EOF, functions return -1 with errno clear.
  */
 
+/*
+ * Fill-in the supplied version components with the Lustre version
+ * found in /sys.
+ *
+ * Returns -1 or less on error, 0 or greater on success.
+ */
+int
+sys_fs_lustre_version (pctx_t ctx, int *major, int *minor, int *patch,
+                       int *fix)
+{
+    int ret = -1;
+    char version[256];
+
+    if ((ret = proc_openf (ctx, PROC_FS_LUSTRE_VERSION)) < 0)
+        goto done;
+
+    if ((ret = proc_gets (ctx, NULL, version, sizeof (version))) < 0)
+        goto close;
+
+    /* first, get the numbers which must be present in a version string */
+    if ((ret = sscanf (version, "%d.%d.%d", major, minor, patch)) != 3) {
+        errno = EIO;
```

Comment:
Why not preserve the error code set by sscanf?

Reply:
`sscanf` won't always have an error code set if it manages to read 1 or 2 items from the string. That is why `errno` is being explicitly set to `EIO`.